### PR TITLE
Add an output directory option to genvk.py.

### PR DESCRIPTION
**Pull request by null77**
Currently for the threading utils and parameter checker generator,
the script is hard-coded to write to the current directory. Add
an option to add a custom output directory.

This allows me to use these scripts within the ANGLE build.

### CODE REVIEW

File: generator.py [hunk 1]
```diff
@@ -561,7 +565,10 @@ def beginFile(self, genOpts):
         # Open specified output file. Not done in constructor since a
         # Generator can be used without writing to a file.
         if (self.genOpts.filename != None):
-            self.outFile = open(self.genOpts.filename, 'w')
+            if (self.genOpts.genDirectory != None):
+                self.outFile = open(self.genOpts.genDirectory + '/' + self.genOpts.filename, 'w')
```
Comment (dustin-lunarg) [inline generator.py]: Can you change this to use os.path.join()?

Comment (null77) [inline generator.py]: Done, good catch.

**Comment by null77** [comment]
Uploaded an updated CL with os.path.join instead of string concatenation.

**Comment by dustin-lunarg** [comment]
Cherry-picked to master and pushed.

**Comment by null77** [comment]
Thanks!

**Pull request closed without merging**